### PR TITLE
Fix various bugs

### PR DESCRIPTION
**Pull request by Yelinz**
- Reset query parameters when reentering route
- Change tag suggestion filter to `istartswith`
- only show mark filters when they are set

### CODE REVIEW

File: addon/services/tags.js [hunk 1]
```diff
@@ -39,7 +39,7 @@ export default class TagsService extends Service {
         const tagId = dasherize(tagInput.trim());
         const existing = (
           await this.store.query("tag", {
-            filter: { nameExact: tagId },
+            filter: { name: tagId },
```
Comment (czosel) [inline addon/services/tags.js:42]: this doesn't seem to make sense, let's revert it

File: addon/components/tag-filter.js [hunk 2]
```diff
@@ -20,6 +21,26 @@ export default class TagFilterComponent extends Component {
...
+          marks.length > 0
+        ) {
+          marks.forEach((mark) => acc.add(mark.id));
+        }
+
+        return acc;
+      }, new Set());
+
+    return this.marks.marks.records?.filter((mark) => activeMarks.has(mark.id));
+  });
```
Comment (czosel) [inline addon/components/tag-filter.js:42]: let's pass the currently displayed documents to the tag filter, and check those for used marks (and tags as well!)

File: tests/integration/components/document-card-test.js [hunk 3]
```diff
@@ -2,7 +2,7 @@ import Service from "@ember/service";
 import { render, click } from "@ember/test-helpers";
 import { setupRenderingTest } from "dummy/tests/helpers";
 import { hbs } from "ember-cli-htmlbars";
-import fileSaver from "file-saver";
+import * as fileSaver from "file-saver";
```
Comment (czosel) [inline tests/integration/components/document-card-test.js:5]: was this broken before?

**Review by czosel — DISMISSED** [review]
Otherwise LGTM

File: tests/integration/components/document-card-test.js (hunk 3 above)
Comment (Yelinz) [reply tests/integration/components/document-card-test.js:5]: yes

Comment (Yelinz) [reply tests/integration/components/document-card-test.js:5]: current test failures on master are from this

**Pull request merged**